python/test/unit/function/test_function.py

```diff
@@ -365,3 +367,52 @@ def expr_eval(values, x, cell_idx):
 
     f.interpolate(f1)
     assert (f.vector().get_local() == 1.0).all()
+
+
+@skip_if_complex
+def test_cffi_expression(V):
+
+    code_h = """
+    void eval(double* values, const double* x, const int64_t* cell_idx,
+            int num_points, int value_size, int gdim, int num_cells);
+    """
+
+    code_c = """
+    void eval(double* values, const double* x, const int64_t* cell_idx,
+            int num_points, int value_size, int gdim, int num_cells)
+    {
+        for (int i = 0; i < num_points; ++i)
+        {
+            values[i*value_size + 0] = x[i*gdim + 0] + x[i*gdim + 1];
+        }
+    }
+    """
+    module = "_expr_eval"
+
+    # Build the kernel
+    ffi = cffi.FFI()
+    ffi.set_source(module, code_c)
+    ffi.cdef(code_h)
+    ffi.compile()
+
+    # Import the compiled kernel
+    kernel_mod = importlib.import_module(module)
+    ffi, lib = kernel_mod.ffi, kernel_mod.lib
+
+    # Get pointer to the compiled function
+    eval_ptr = ffi.cast("uintptr_t", ffi.addressof(lib, "eval"))
+
+    # Handle C func address by hand
+    ex1 = Expression(int(eval_ptr))
```

Comment:
Is `int` required?

Reply:
Oddly enough it is. `ffi.cast("uintptr_t")` cast to cffi type `<class '_cffi_backend.CData'>`. With your latest changes on how int is casted to std::function it just doesn't work.

The old way that we have in `set_cell_tabulate` (and we had in expression) accepts also these cffi uintptr_t types.

Reply:
I see now, pybind tries to be clever and binds cffi uintptr to constructor with `std::function` directly. I'll update PR asap.

Reply:
OK, accepting `std::function` in pybind11 argument is nice but dangerous.
It accepts `cffi uintptr_t` objects and then tries to call it as a method inside evaluation. On the other hand it allows to construct Expression directly on pure python method.

I propose to keep `int()` as is in this PR. Without the `int()` there is an exception, so no ill-defined behaviour could happen.